cohortextractor/definition/definition.py

```diff
@@ -1,8 +1,31 @@
+from ..query_interface import Variable
 from .base import cohort_registry
 
 
 class Cohort:
-    ...
+    """Represents the cohort of patients in a study."""
+
+    def __init__(self) -> None:
+        self.population = NotImplemented
+
+    def set_population(self, population: Variable) -> None:
+        """Set the population variable for this cohort."""
+        self.add_variable("population", population)
+        if not self.population.reduce_function.__name__ == "exists":
+            raise ValueError(
+                "Population variable must return a boolean. Did you mean to use `make_one_row_per_patient(exists)`?"
+            )
```

Comment:
I think it would make sense to do this check before the call to `add_variable`.

Reply:
I couldn't decide whether to do it before or after, but I'd thought it maybe made sense to raise the exception if it wasn't a Variable at all first (which happens in `add_variable`?

Reply:
Ah yes, good point!